ksqldb-engine/src/main/java/io/confluent/ksql/engine/KsqlEngine.java

```diff
@@ -213,6 +213,11 @@ public boolean hasActiveQueries() {
     return !primaryContext.getQueryRegistry().getPersistentQueries().isEmpty();
   }
 
+  public void restartStreamsRuntime() {
```

Comment:
Why do we need this at this level? would it be better to just call this whenever the alter system is called? Or should we leave them decoupled? WDYT

Reply:
I think we need to seperate it as we call the method in the `ValidatedCommandFactory` for validation purposes. If we didn't the streams runtime.

Reply:
ah yeah thats a good point